templates/src/app.js

```diff
@@ -1,31 +1,65 @@
-var dotenv = require("dotenv");
-var express = require("express");
-var morgan = require("morgan");
+const dotenv = require("dotenv");
+const express = require("express");
+const morgan = require("morgan");
+const dbDatasource = require("./db");
+<%if eq (index .Params `graphql`) "yes" %>const { ApolloServer } = require("apollo-server-express");
 
-var { connect } = require("./db");
+const typeDefs = require("./app/graphql/schema");
+const { createStore } = require("./app/graphql/utils");
+const resolvers = require("./app/graphql/resolvers");
+const LaunchAPI = require('./app/graphql/datasources/launch');
+const UserAPI = require('./app/graphql/datasources/user');
+const isEmail = require('isemail');<% end %>
+<%if eq (index .Params `fileUploads`) "yes" %>const fileRoutes = require("./app/file");<% end %>
 const statusRoutes = require("./app/status");
-<%if eq (index .Params `fileUploads`) "yes" %>const fileRoutes = require("./app/file");
-<% end %><%if eq (index .Params `userAuth`) "yes" %>const authRoutes = require("./app/auth");
-<% end %>
+<%if eq (index .Params `userAuth`) "yes" %>const authRoutes = require("./app/auth");<% end %>
+
 dotenv.config();
-var app = express();
+const app = express();
 app.use(morgan("combined"));
+app.use(express.json());
+app.use(express.urlencoded({ extended: false }));
+
+<%if eq (index .Params `graphql`) "yes" %>
+const store = createStore();
+
+const server = new ApolloServer({
+  context: async ({ req }) => {
+    // simple auth check on every request
+    const auth = req.headers && req.headers.authorization || '';
+    const email = Buffer.from(auth, 'base64').toString('ascii');
+    if (!isEmail.validate(email)) return { user: null };
+    // find a user by their email
+    const users = await store.users.findOrCreate({ where: { email } });
+    const user = users && users[0] || null;
+    return { user: { ...user.dataValues } };
+  },
+  typeDefs,
+  resolvers,
+  dataSources: () => ({
+    launchAPI: new LaunchAPI(),
+    userAPI: new UserAPI({ store })
+  }),
+});
+server.applyMiddleware({ app });<% end %>
+
+<%if eq (index .Params `fileUploads`) "yes" %>app.use("/file", fileRoutes);<% end %>
```

Comment:
Could you add the functionality from the file and auth routes to the gql schema as well? We will want to expose the same functionality regardless of the api type they choose

Reply:
Sure. I can make it.

Reply:
Wrapped /file and /auth APIs with graphql.